main/src/main/java/cgeo/geocaching/SplashActivity.java

```diff
@@ -29,6 +30,9 @@ public void onCreate(final Bundle savedInstanceState) {
                 // new install, base folder missing or folder migration needed => run installation wizard
                 intent = new Intent(this, InstallWizardActivity.class);
                 intent.putExtra(InstallWizardActivity.BUNDLE_MODE, firstInstall ? InstallWizardActivity.WizardMode.WIZARDMODE_DEFAULT.id : InstallWizardActivity.WizardMode.WIZARDMODE_MIGRATION.id);
+            } else if (Settings.getBackupRestoreRequested() > 0) {
+                // if any pending backup requests are found: start backup
+                intent = new Intent(this, BackupActivity.class);
```

Comment:
Wouldn't it be enpough ti synchronously execute backup/restore here? Is a new activity really needed?

Reply:
Possibly yes, but I wanted to keep backup details clear from `SplashActivity` (which is not much more than some switch on startup), and also `SplashActivity` uses a very reduced and special theme, whereas `BackupActivity` uses the regular c:geo theme.